lib/mapUtils.js

```diff
@@ -292,3 +298,131 @@ export function openDirections(latitude, longitude, storeName) {
     alwaysIncludeGoogle: true,
   });
 }
+
+export function findStoreDistance(region, store) {
+  const distanceMeters = getDistance(region, store);
+  // Convert distance to 'x.xx' form, in miles units
+  const distance = convertDistance(distanceMeters, 'mi').toFixed(2);
+  return distance;
+}
+
+export async function orderStoresByDistance(region, inputStores) {
+  const sortedStores = [];
+  // Display distance in the StoreList
+  inputStores.forEach((store) => {
+    const currStore = store;
+    currStore.distance = findStoreDistance(region, store);
+    sortedStores.push(currStore);
+  });
+  // sorts in place
+  sortedStores.sort(function compare(a, b) {
+    return a.distance - b.distance;
+  });
+  return sortedStores;
+}
+
+export function findDefaultStore(stores) {
+  const defaultStore = stores.find((store) => {
+    return store.id === RecordIds.defaultStoreId;
+  });
+
+  const defaultRegion = {
+    latitude: defaultStore.latitude,
+    longitude: defaultStore.longitude,
+    ...deltas,
+  };
+  return { defaultStore, defaultRegion };
+}
+
+export function useCurrentLocation(setInitialLocation, setLocationPermissions) {
+  useEffect(() => {
+    const getCurrentLocation = async () => {
```

Comment:
Nit: to flatten, you could move this outside of the effect and only call `getCurrentLocation` within the effect.